src/librustc_trans/diagnostics.rs

```diff
@@ -12,6 +12,96 @@
 
 register_long_diagnostics! {
 
+E0510: r##"
+`return_address` was used in an invalid context. Erroneous code example:
+
+```
+extern "rust-intrinsic" {
+    fn return_address() -> *const u8;
+}
+
+pub unsafe fn by_value() -> i32 {
+    let _ = return_address();
+    // error: invalid use of `return_address` intrinsic: function does
+    //        not use out pointer
+    0
+}
+```
+
+Returned values are stored in registers. In the case where the returned
```

Comment:
Return values may be stored in a return register(s) or written into a so-called out pointer. In case the returned value is big enough to not fit into the return register(s), the compiler will return value by writing it into space allocated in the caller’s stack frame.

You probably also should note that notion of “big enough” is target-ABI-dependent and in general not portable or future-proof.